File: AN_2/SEM_1/MAP/lab2/src/tests/DelayTaskRunnerTest.java
package tests;

import Enum.Strategy;
import model.MessageTask;
import decorator.DelayTaskRunner;
import decorator.PrinterTaskRunner;
import decorator.StrategyTaskRunner;

public class DelayTaskRunnerTest
{
    public static void delayTaskRunnerTest(Strategy strategy, MessageTask[] messageTasks)
    {
        StrategyTaskRunner strategyTaskRunner = new StrategyTaskRunner(strategy);
        DelayTaskRunner delayTaskRunner = new DelayTaskRunner(strategyTaskRunner);
        PrinterTaskRunner printerTaskRunner = new PrinterTaskRunner(delayTaskRunner);
        printerTaskRunner.addTask(messageTasks[0]);
        printerTaskRunner.addTask(messageTasks[1]);
        printerTaskRunner.addTask(messageTasks[2]);
        printerTaskRunner.executeAll();
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Tests/DelayTaskRunnerTest.cs
using Enum;
using Model;
using Decorator;

namespace Tests
{
    public class DelayTaskRunnerTest
    {
        public static void RunDelayTaskRunnerTest(Strategy strategy, MessageTask[] messageTasks)
        {
            var strategyTaskRunner = new StrategyTaskRunner(strategy);
            var delayTaskRunner = new DelayTaskRunner(strategyTaskRunner);
            var printerTaskRunner = new PrinterTaskRunner(delayTaskRunner);
            printerTaskRunner.AddTask(messageTasks[0]);
            printerTaskRunner.AddTask(messageTasks[1]);
            printerTaskRunner.AddTask(messageTasks[2]);
            printerTaskRunner.ExecuteAll();
        }
    }
}
